.github/workflows/deploy-to-k8s.yaml

```diff
@@ -0,0 +1,68 @@
+name: deploy-to-k8s
+on:
+  workflow_call:
+    inputs:
+      KUBECTL_VERSION:
+        description: 'kubectl version'
+        type: string
+        required: true
+      NAMESPACE:
+        description: 'namespace of the k8s cluster'
+        type: string
+        required: true
+      SECRET_FROM_ENV_PATH:
+        description: 'path to the k8s secret from env yaml file'
+        type: string
+      SECRET_NAME:
+        description: 'name for the k8s secret'
+        type: string
+      MANIFEST_OR_FOLDER_PATH:
+        description: 'path to the manifest yaml file or folder'
+        type: string
+        required: true
+    secrets:
+      KUBE_CONFIG_BASE64:
+        description: 'kube config file in base64 encoded'
+        required: true
+
+jobs:
+  deploy-k8s:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout caller repo
+        uses: actions/checkout@v4
+      - name: Setup kubectl
```

Comment:
I'd consider adding a sanity check step here, which just makes sure that all the input paths are correct before trying to deploy anything:

```yaml
- name: Check Paths
  run: ls ${{ inputs.SECRET_FROM_ENV_PATH}} ${{ github.workspace}}/${{ inputs.MANIFEST_OR_FOLDER_PATH }} || echo "Check paths for env file and manifest/folder" && exit 1
```

Reply:
Added, thanks!